wp-content/plugins/tapestry/controller/class.user-controller.php

```diff
@@ -0,0 +1,105 @@
+<?php
+/**
+ * Add/update/retrieve User progress
+ * 
+ */
+class UserController {
```

Comment:
You'll need to use a prefix when naming any classes or global variables and functions. Anything inside your class is fine since it's scope is limited to the class, but anything outside the class, including the class name itself should have a prefix. You can use `Tapestry` or `tapestry` as the prefix for everything. So for example here your controller should be called `TapestryUserController`. See https://developer.wordpress.org/plugins/plugin-basics/best-practices/#avoid-naming-collisions

Reply:
Changed to TapestryUserController